nvflare/app_common/homomorphic_encryption/he_model_decryptor.py

```diff
@@ -130,16 +130,16 @@ def _process(self, shareable: Shareable, fl_ctx: FLContext):
         self.log_info(fl_ctx, "Running decryption...")
         dxo = from_shareable(shareable)
 
-        encrypted_algo = dxo.get_meta_prop(key=MetaKey.PROCESSED_ALGORITHM, default=None)
-        if encrypted_algo != he.HE_ALGORITHM_CKKS:
-            self.log_error(fl_ctx, "shareable is not HE CKKS encrypted")
-            return shareable
-
         encrypted_layers = dxo.get_meta_prop(key=MetaKey.PROCESSED_KEYS, default=None)
         if not encrypted_layers:
             self.log_warning(fl_ctx, "dxo does not contain PROCESSED_KEYS (do nothing)")
             return shareable
 
+        encrypted_algo = dxo.get_meta_prop(key=MetaKey.PROCESSED_ALGORITHM, default=None)
+        if encrypted_algo != he.HE_ALGORITHM_CKKS:
+            self.log_error(fl_ctx, "shareable is not HE CKKS encrypted")
+            return shareable
```

Comment:
Hi @holgerroth , what is the difference for this change?

Reply:
This addresses issue https://github.com/NVIDIA/NVFlare/issues/320. Now, the printout by the decryptor in the first round (when it get's unencrypted weights from the server) will show
` HEModelDecryptor - WARNING - [run=1, peer=secure_project, peer_run=1, task_name=train, task_id=785a7e00-5d79-4c01-af94-f3d602e837bb]: dxo does not contain PROCESSED_KEYS (do nothing)`
instead of previously
`2022-03-10 16:40:27,221 - HEModelDecryptor - ERROR - [run=1, peer=secure_project, peer_run=1, task_name=train, task_id=abc44040-ebf5-4474-8ea6-9f2a74f0badb]: shareable is not HE CKKS encrypted` 
which was confusing.